src/main/java/com/pngencoder/PngEncoder.java

```diff
@@ -38,38 +57,88 @@ public PngEncoder() {
                 null);
     }
 
+    /**
+     * Returns a new PngEncoder which has the same configuration as this one except {@code bufferedImage}.
+     * The new PngEncoder will use the provided {@code bufferedImage}.
+     *
+     * @param bufferedImage input image
+     * @return a new PngEncoder
+     */
     public PngEncoder withBufferedImage(BufferedImage bufferedImage) {
         return new PngEncoder(bufferedImage, compressionLevel, multiThreadedCompressionEnabled, srgbRenderingIntent);
     }
 
+    /**
+     * Returns a new PngEncoder which has the same configuration as this one except {@code compressionLevel}.
+     * The new PngEncoder will use the provided {@code compressionLevel}.
+     *
+     * @param compressionLevel input image (must be between -1 and 9 inclusive)
+     * @return a new PngEncoder
+     */
     public PngEncoder withCompressionLevel(int compressionLevel) {
         return new PngEncoder(bufferedImage, compressionLevel, multiThreadedCompressionEnabled, srgbRenderingIntent);
     }
 
+    /**
+     * Returns a new PngEncoder which has the same configuration as this one except {@code multiThreadedCompressionEnabled}.
+     * The new PngEncoder will use the provided {@code multiThreadedCompressionEnabled}.
+     *
+     * @param multiThreadedCompressionEnabled when {@code true}, multithreaded compression will be used
+     * @return a new PngEncoder
+     */
     public PngEncoder withMultiThreadedCompressionEnabled(boolean multiThreadedCompressionEnabled) {
         return new PngEncoder(bufferedImage, compressionLevel, multiThreadedCompressionEnabled, srgbRenderingIntent);
     }
 
+    /**
+     * Returns a new PngEncoder which has the same configuration as this one except {@code srgbRenderingIntent}.
+     * The new PngEncoder will use the provided {@code srgbRenderingIntent}.
+     *
+     * @param srgbRenderingIntent todo
+     * @return a new PngEncoder
+     */
     public PngEncoder withSrgbRenderingIntent(PngEncoderSrgbRenderingIntent srgbRenderingIntent) {
         return new PngEncoder(bufferedImage, compressionLevel, multiThreadedCompressionEnabled, srgbRenderingIntent);
     }
 
+    /**
+     * Returns the {@code BufferedImage} provided at construction time.
+     * @return the {@code BufferedImage} provided at construction time
+     */
     public BufferedImage getBufferedImage() {
         return bufferedImage;
     }
 
+    /**
```

Comment:
I think documenting simple getters are overkill. If there is documentation on a getter, it will make me quickly assume that there is something more complex going on, breaking my flow. It would take a little extra time to realize that it is in fact simple.

Reply:
Yeah, I can see that. Thanks for the feedback.

Reply:
I agree that these feel a bit overkill, especially since they also duplicate the same text twice.